satpy/tests/reader_tests/test_seviri_l1b_calibration.py

```diff
@@ -93,108 +98,214 @@
-        assertNumpyArraysEqual(result, VIS008_REFLECTANCE)
+        """Test conversion from radiance to reflectance."""
+        result = self.algo.vis_calibrate(VIS008_RADIANCE,
+                                         VIS008_SOLAR_IRRADIANCE)
+        np.testing.assert_allclose(result, VIS008_REFLECTANCE)
         self.assertTrue(result.sun_earth_distance_correction_applied)
 
-    def tearDown(self):
-        pass
+
+class TestSeviriCalibrationHandler:
+    """Unit tests for SEVIRI calibration handler."""
+
+    def test_init(self):
+        """Test initialization of the calibration handler."""
+        with pytest.raises(ValueError):
+            SEVIRICalibrationHandler(
+                platform_id=None,
+                channel_name=None,
+                coefs=None,
+                calib_mode='invalid',
+                scan_time=None
+            )
+
+    @pytest.fixture(name='calib')
+    def calib(self):
+        """Provide a calibration handler."""
+        return SEVIRICalibrationHandler(
+            platform_id=324,
+            channel_name='IR_108',
+            coefs={
+                'coefs': {
+                    'NOMINAL': {
+                        'gain': 10,
+                        'offset': -1
+                    },
+                    'GSICS': {
+                        'gain': 20,
+                        'offset': -2
+                    },
+                    'EXTERNAL': {}
+                },
+                'radiance_type': 1
+            },
+            calib_mode='NOMINAL',
+            scan_time=None
+        )
+
+    def test_calibrate_exceptions(self, calib):
+        """Test exceptions raised by the calibration handler."""
+        with pytest.raises(ValueError):
+            calib.calibrate(None, 'invalid')
+
+    @pytest.mark.parametrize(
+        ('calib_mode', 'ext_coefs', 'expected'),
+        [
+            ('NOMINAL', {}, (10, -1)),
+            ('GSICS', {}, (20, -40)),
+            ('GSICS', {'gain': 30, 'offset': -3}, (30, -3)),
+            ('NOMINAL', {'gain': 30, 'offset': -3}, (30, -3))
+        ]
+    )
+    def test_get_gain_offset(self, calib, calib_mode, ext_coefs, expected):
+        """Test selection of gain and offset."""
+        calib.calib_mode = calib_mode
+        calib.coefs['coefs']['EXTERNAL'] = ext_coefs
+        coefs = calib.get_gain_offset()
+        assert coefs == expected
+
+
+class TestFileHandlerCalibrationBase:
+    """Base class for file handler calibration tests."""
+
+    platform_id = 324
+    gains_nominal = np.arange(1, 13)
+    offsets_nominal = np.arange(-1, -13, -1)
+    gains_gsics = np.arange(0.1, 1.3, 0.1)
+    offsets_gsics = np.arange(-0.1, -1.3, -0.1)
+    radiance_types = 2 * np.ones(12)
+    scan_time = datetime(2020, 1, 1)
+    external_coefs = {
+        'VIS006': {'gain': 10, 'offset': -10},
+        'IR_108': {'gain': 20, 'offset': -20}
+    }
+    spectral_channel_ids = {'VIS006': 1, 'IR_108': 9}
+    expected = {
```

Comment:
I thought it's easier to specify this as a dictionary here instead of making it an "expected" parameter of `test_calibrate`